src/metabase/db/custom_migrations/metrics_v2.clj

```diff
@@ -0,0 +1,158 @@
+  If and only if the collection doesn't exists and the option `create?` is truthy,
+  the collection is created."
+  ([]
+   (get-metric-migration-collection nil))
+  ([{:keys [create?]}]
+   (let [coll-name "Migrated Metrics v1"
+         slug (u/slugify coll-name 510)
+         desc "This collection and the metrics in it were automatically generated by the migration to v50."]
+     (or (t2/select-one :collection :name coll-name :slug slug :description desc)
+         (when create?
+           (let [collection (t2/insert-returning-instance!
+                             :collection
+                             {:name coll-name, :slug slug, :description desc})]
+             (let [all-users-group-id (t2/select-one-fn :id :permissions_group :name "All Users")]
+               (t2/insert! :permissions {:object   (format "/collection/%s/read/" (:id collection))
+                                         :group_id all-users-group-id}))
+             collection))))))
+
+(defn- add-metric-id
+  "Add `id` (the ID of the metric being migrated) to `description`.
+  This ID is parsed from the description during rollback. See [[split-metric-id]]."
+  [description id]
+  (str description " (Migrated from metric " id ".)"))
+
+(defn- convert-metric-v2
+  "Convert a metric v2 question from `metric-v1` defined on a table in the
+  database with ID `db-id`.
+  The :collection_id property is not set and the question is not persisted."
+  [metric-v1 db-id]
+  (let [definition (json/parse-string (:definition metric-v1) true)
+        dataset-query {:type :query
+                       :database db-id
+                       :query definition}]
+    (-> metric-v1
+        (select-keys [:archived :created_at :creator_id :database_id
+                      :description :name :table_id])
+        (update :description add-metric-id (:id metric-v1))
+        (assoc :dataset_query (json/generate-string dataset-query)
+               :enable_embedding false
+               :query_type "query"
+               :type "metric"
+               :parameters "[]"
+               :parameter_mappings "[]"
+               :visualization_settings "{}"
+               :display "line"))))
+
+(defn- create-metric-v2
+  "Create and persist a metric v2 question from `metric-v1` in `metric-v2-coll`."
+  [metric-v1 metric-v2-coll]
+  (let [db-id (t2/select-one-fn :db-id :metabase_table (:table_id metric-v1))
+        card (-> metric-v1
+                 (convert-metric-v2 db-id)
+                 (assoc :collection_id (:id metric-v2-coll)
+                        :updated_at (Instant/now)))]
+    (t2/insert-returning-instance! :report_card card)))
+
+(defn- metric-ref->id
+  "If `expr` is a metric reference expression, return the ID of the referenced metric.
+  Returns nil otherwise."
+  [expr]
+  (when (and (vector? expr)
+             (= (u/lower-case-en (get expr 0)) "metric"))
+    (let [id (get expr 1)]
+      ;; GA metrics with string references are ignored
```

Comment:
Should we rename these tags to something like `[:legacy-metric "..."]`? I'm trying to weigh the harm of strings showing up where v2 expects id

Reply:
Well, such questions are broken and stay broken. The worst that can happen that they break in a more obscure way, right?

Reply:
Yeah, I guess a preprocess step that looks for strings to throw nice errors is no different than one that looks for the tag.

Reply:
If we drop support for GA metrics then we should probably just drop support for GA as a whole, since GA queries require metrics. You can't run a GA query without metrics

Reply:
[Slack thread](https://metaboat.slack.com/archives/C04DN5VRQM6/p1715888743074829)

Reply:
Ok the conclusion is that we will remove the GA driver before we ship 50 https://github.com/metabase/metabase/issues/42792

Reply:
@camsaul, what change does that require here? Removing the comment and the `int?` check?

Reply:
I think you can just leave it as-is, like you said these queries aren't going to work anymore anyway once we remove the GA driver. I don't think we will touch any queries as part of that project so you should continue to ignore them